pkg/manager/member/scaler.go

```diff
@@ -104,3 +113,45 @@ func ordinalPVCName(memberType v1alpha1.MemberType, setName string, ordinal int3
 func ordinalPodName(memberType v1alpha1.MemberType, tcName string, ordinal int32) string {
 	return fmt.Sprintf("%s-%s-%d", tcName, memberType, ordinal)
 }
+
+// scaleOne calculates desired replicas and delete slots from actual/desired
+// stateful sets by allowing only one pod to be deleted or created
+// it returns following values:
+// - scaling:
+//   - 0: no scaling required
+//   - 1: scaling out
+//   - -1: scaling in
+// - ordinal: pod ordinal to create or delete
+// - replicas/deleteSlots: desired replicas and deleteSlots by allowing only one pod to be deleted or created
+func scaleOne(actual *apps.StatefulSet, desired *apps.StatefulSet) (scaling int, ordinal int32, replicas int32, deleteSlots sets.Int32) {
+	actualDesiredPodOrdinals := helper.GetPodOrdinals(*actual.Spec.Replicas, actual)
+	desiredDesiredPodOrdinals := helper.GetPodOrdinals(*desired.Spec.Replicas, desired)
+	additions := desiredDesiredPodOrdinals.Difference(actualDesiredPodOrdinals)
+	deletions := actualDesiredPodOrdinals.Difference(desiredDesiredPodOrdinals)
```

Comment:
Is the order of the diff result guaranteed? or random?

Reply:
`List()` will sort the set members in ascending order